src/transformStateWithClones.js

```diff
@@ -6,8 +7,69 @@
  *
  * @return {Object[]}
  */
+// function transformStateWithClones(state, actions) {
+//   // write code here
+//   const table = {
+//     ...state,
+//   };
+//   const result = [];
+
+//   for (const action of actions) {
+//     if (action.type === 'clear') {
+//       for (const delet in table) {
+//         delete table[delet];
+//       }
+//       console.log(table);
+//     } else if (action.type === 'removeProperties') {
+//       for (const remove of action.keysToRemove) {
+//         delete table[remove];
+//       }
+//     } else {
+//       Object.assign(table, action.extraData);
+//     }
+
+//     result.push({ ...table });
+//   }
+
+//   return result;
+// }
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let currentState = {
+    ...state,
+  }
+
+  for (const action of actions) {
+    let copyOfState = {
+      ...currentState,
+    };
+
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(copyOfState, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const remove of action.keysToRemove) {
+          delete copyOfState[remove];
+        }
+        break;
+
+      case 'clear':
+        copyOfState = {};
+        break;
+
+      default:
+        return `Unknown action type: ${action.type}`;
+    }
+
+    currentState = { ...copyOfState };
```

Comment:
[DONT REPEAT YOURSELF] - Since you are already creating a copy of the state with `let copyOfState` and then pushing `copyOfState` to the result array, there is no need to reassign `currentState` with `copyOfState` here. This line can be removed to avoid unnecessary operations.